PyViCare/PyViCareService.py

```diff
@@ -45,13 +41,21 @@ def __init__(self, oauth_manager: AbstractViCareOAuthManager, accessor: ViCareDe
         self.roles = roles
 
     def getProperty(self, property_name: str) -> Any:
-        url = buildGetPropertyUrl(
-            self.accessor, property_name)
+        url = self.buildGetPropertyUrl(property_name)
         return self.oauth_manager.get(url)
+        
+    def buildGetPropertyUrl(self, property_name):
+        if self._isGateway():
+            return f'/features/installations/{self.accessor.id}/gateways/{self.accessor.serial}/features/{property_name}'
+        return f'/features/installations/{self.accessor.id}/gateways/{self.accessor.serial}/devices/{self.accessor.device_id}/features/{property_name}'
+
 
     def hasRoles(self, requested_roles) -> bool:
         return hasRoles(requested_roles, self.roles)
 
+    def _isGateway(self) -> bool:
+        return self.hasRoles(["type:gateway;VitoconnectOpto1"])
```

Comment:
Is it fine to just check the roles here and not the model as in `asAutoDetectDevice`?

Reply:
yes, I would keep that until, we have a second device which does not work correctly

Reply:
Probably `Heatbox2_SRC` and `E3_TCU10_x07` should be handled as Gateways as well.